packages/ui/react-ui-grid/src/Grid/Grid.tsx

```diff
@@ -106,22 +105,35 @@ const GRID_CONTENT_NAME = 'GridContent';
 
 const GridContent = forwardRef<NaturalDxGrid, GridScopedProps<GridContentProps>>((props, forwardedRef) => {
   const { id, editing, setEditBox, setEditing } = useGridContext(GRID_CONTENT_NAME, props.__gridScope);
-  const dxGrid = useForwardedRef(forwardedRef);
+  const [dxGrid, setDxGridInternal] = useState<NaturalDxGrid | null>(null);
+
+  const setDxGrid = useCallback(
```

Comment:
maybe similar note here

Reply:
todo: factor out hook?

Reply:
I‘m not sure that’s necessarily motivated this early, I can’t predict yet how frequently we’ll need this pattern. Will leave a note for sure though.